core/base/projectionFromTable/ProjectionFromTable.h

```diff
@@ -0,0 +1,132 @@
+/// \ingroup base
+/// \class ttk::ProjectionFromTable
+/// \author Mathieu Pont <mathieu.pont@lip6.fr>
+/// \date 2022.
+///
+/// This module defines the %ProjectionFromTable class that
+/// projects on a surface points in a vtkTable.
+///
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#include <Geometry.h>
+#include <Triangulation.h>
+
+namespace ttk {
+
+  /**
+   * The ProjectionFromTable class provides methods that
+   * projects on a surface points in a vtkTable.
+   */
+  class ProjectionFromTable : virtual public Debug {
+
+  protected:
+  public:
+    ProjectionFromTable();
+
+    int preconditionTriangulation(AbstractTriangulation *triangulation) {
+      // Pre-condition functions.
+      if(triangulation) {
+      }
+      return 0;
+    }
+
+    template <class triangulationType, class xDataType, class yDataType>
+    void computeInputPoints(
+      const triangulationType *triangulation,
+      std::vector<std::tuple<int, double, double>> &surfaceValues,
+      std::array<const long, 2> &surfaceDim,
+      const xDataType *const tableXValues,
+      const yDataType *const tableYValues,
+      const size_t nTableValues,
+      std::vector<std::vector<double>> &inputPoints) {
+      unsigned int noPoints = nTableValues;
+      inputPoints = std::vector<std::vector<double>>(
+        noPoints, std::vector<double>(3, 0.0));
+
+      // Find quadrant of each point
+      // 0 --- 2
+      // |     |
+      // 1 --- 3
+      std::vector<std::vector<int>> quadPoints(noPoints, std::vector<int>(4));
+      for(unsigned int i = 0; i < noPoints; ++i) {
+        int i0 = 0, i1 = 0;
+        for(unsigned int j = 0; j < surfaceDim[0]; ++j)
+          if(std::get<1>(surfaceValues[j * surfaceDim[1]]) < tableXValues[i])
+            i0 = j;
+        for(unsigned int j = 0; j < surfaceDim[1]; ++j)
+          if(std::get<2>(surfaceValues[j]) < tableYValues[i])
+            i1 = j;
+        quadPoints[i][0] = i0 * surfaceDim[1] + i1 + 1;
+        quadPoints[i][1] = i0 * surfaceDim[1] + i1;
+        quadPoints[i][2] = (i0 + 1) * surfaceDim[1] + i1 + 1;
+        quadPoints[i][3] = (i0 + 1) * surfaceDim[1] + i1;
+      }
+
+      // Iterate through each point
+      std::vector<std::vector<double>> coef(
+        noPoints, std::vector<double>(4, 0.0));
```

Comment:
Replace with `std::vector<std::array<double, 4>> coef(nPoints)`? (`std::array`s are 0-initialized)

Reply:
Done in the next commit!